src/adapter/src/explain_new/qgm/dot.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! DOT format `EXPLAIN` support for `QGM` structures.
+
+use mz_repr::explain_new::DisplayDot;
+
+/// A very naive way of representing a
+/// [`mz_repr::explain_new::ExplainFormat::Dot`] explanation for an
+/// [`mz_sql::query_model::Model`].
+///
+/// Because `mz_sql::query_model::dot::DotGenerator::generate` is
```

Comment:
I guess here we add a todo for deriving the required attributes for the model in `explain_dot` so that `mz_sql::query_model::dot::DotGenerator::generate` can be immutable.